src/core/hooks/use-form-controller/atom-cache.ts

```diff
@@ -41,15 +43,18 @@ export class FieldStateAtomCache<Values extends object, Err> {
     field: FieldDescriptor<T, Err>
   ): FieldStateAtom<T> {
     const lens = impl(field).__lens;
+    const initialValue = lens.get(this.formtsState.initialValues);
+    const fieldValueAtom = Atom.entangle(this.formtsState.values, lens);
 
     return Atom.fuse(
-      (value, touched, formSubmitted) => ({
+      (value, changed, touched, formSubmitted) => ({
         value,
+        changed,
```

Comment:
we can compute `changed` here based on `value`, without the need to introduce a new Atom out of `.fuse` (line 57), by simply comparing `value` and `initialValue` in the combinator body 
```suggestion
        changed: !deepEqual(value, initialValue)
```

Reply:
I'd keep it - there is a small difference - as a separate atom deepEqual will only run on value changes, but when put in the combinator fn it would run on change of any dependency (and it is somewhat heavy operation). I think we will need to remove `FieldDependenciesAtom` and move error and validating state computations here in the future as part #127